# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1318868586 14400
#      Mon Oct 17 12:23:06 2011 -0400
# Node ID d6aaa28dff504647fc222c9423946ea1115d75fe
# Parent  6bd38ab573033cf3938e7cd6b36bb994f3c64bbe
don't forget to pass parent class names to objects created by indexing

* ov-class.cc (octave_class::subsref): Pass parent class names to
newly created objects.

diff --git a/src/ov-class.cc b/src/ov-class.cc
--- a/src/ov-class.cc
+++ b/src/ov-class.cc
@@ -489,17 +489,18 @@ octave_class::subsref (const std::string
           // if it has more than one element, to be able to pass through
           // rvalue1 calls.
           if (retval.length () > 1)
             retval = octave_value (retval, true);
         }
       else
         {
           if (type.length () == 1 && type[0] == '(')
-            retval(0) = octave_value (map.index (idx.front ()), class_name ());
+            retval(0) = octave_value (map.index (idx.front ()), class_name (),
+                                      parent_class_name_list ());
           else
             gripe_invalid_index1 ();
         }
     }
 
   return retval;
 }
 
